groups/alteryx-sandbox/ec2_server.tf

```diff
@@ -32,10 +32,10 @@ module "alteryx_server_ec2" {
   vpc_security_group_ids = [
     aws_security_group.ec2.id
   ]
-  subnet_id            = [for sub in data.aws_subnet.alteryx : sub.id][count.index + 1]
+  subnet_id            = [for sub in data.aws_subnet.alteryx : sub.id][count.index]
```

Comment:
As you're now using specific subnets, this can be replaced with a single data lookup for the specific subnet in question. E.g. `data.aws_subnet.alteryx.id`